src/sentry/api/endpoints/accept_organization_invite.py

```diff
@@ -25,14 +27,22 @@ def respond_invalid() -> Response:
     def get_helper(self, request: Request, member_id: int, token: str) -> ApiInviteHelper:
         return ApiInviteHelper(request=request, member_id=member_id, instance=self, token=token)
 
-    def get(self, request: Request, member_id: int, token: str) -> Response:
+    def get(
+        self, request: Request, member_id: int, token: str, organization_slug: Optional[str] = None
+    ) -> Response:
         try:
             helper = self.get_helper(request, member_id, token)
         except OrganizationMember.DoesNotExist:
             return self.respond_invalid()
 
         organization_member = helper.om
-        organization = organization_member.organization
+        organization = helper.organization
+
+        if organization_slug:
+            if organization_slug != organization.slug:
+                return self.respond_invalid()
+        else:
```

Comment:
Should we add the `slug` as a parameter to the `ApiInviteHelper`? That would let us consolidate this logic with the logic that fetches an organization member + organization.